### PR TITLE
[SYCL] Add feature macro for the matrix extension

### PR DESCRIPTION
**Pull request by yubingex007-a11y**

### CODE REVIEW

**Review by Pennycook — APPROVED** [review]

**Comment by yubingex007-a11y** [comment]
ping? @bader

File: sycl/include/CL/sycl/feature_test.hpp [hunk 1]
```diff
@@ -13,6 +13,8 @@ namespace sycl {
 // Feature test macro definitions
 
 #define SYCL_EXT_INTEL_DEVICE_INFO 1
+// TODO: Move feature-test macros to compiler driver.
```
Comment (alexbatashev) [inline sycl/include/CL/sycl/feature_test.hpp:16]: Should this comment apply to all macros?

File: sycl/test/on-device/extensions/matrix-amx-int8-test.cpp [hunk 2]
```diff
@@ -119,6 +120,7 @@ void matrix_multiply_ref(int32_t *A_mem, int32_t *B_mem, int32_t *C_mem, int M,
 }
 
 int main() {
+
```
Comment (alexbatashev) [inline sycl/test/on-device/extensions/matrix-amx-int8-test.cpp:123]: This seems to be an unrelated change.

**Review by alexbatashev — DISMISSED** [review]

**Review by alexbatashev — APPROVED** [review]

**Review by Pennycook — APPROVED** [review]

**Comment by yubingex007-a11y** [comment]
@bader, it seems the merging is still blocked. Do we need appeoval from other reviewers?

**Comment by alexbatashev** [comment]
> @bader, it seems the merging is still blocked. Do we need appeoval from other reviewers?

It says "Merging is blocked" because you don't have permissions to press the merge button. Otherwise, it should be fine. You have all the necessary approvals, and all tests have passed.

**Pull request merged**